### ISSUE TITLE
`global_step` and `epoch` for model loaded from checkpoint are 0

### ISSUE THREAD

**Issue by mbanani** [PyTorchLightning/pytorch-lightning#12819]
## 🐛 Bug

When loading a model using `model.load_from_checkpoint()`, the `global_step` and `epoch` variables are both 0. I would expect those variables to match the stage of the model during training, but maybe there's some context that I am missing. My apologies if this is intended behavior. 

### To Reproduce

```
import torch
import os
import pytorch_lightning
from pytorch_lightning import LightningModule
from pytorch_lightning import Trainer
from torch.utils.data import Dataset, DataLoader


class BoringModel(LightningModule):
    def __init__(self):
        super().__init__()
        self.layer = torch.nn.Linear(32, 2)

    def training_step(self, batch, batch_idx):
        loss = self.layer(batch).sum()
        print(f'Epoch: {self.current_epoch} Global step: {self.global_step}')
        return loss

    def configure_optimizers(self):
        return torch.optim.SGD(self.layer.parameters(), lr=0.1)


class RandomDataset(Dataset):
    def __init__(self, size, num_samples):
        self.len = num_samples
        self.data = torch.randn(num_samples, size)

    def __getitem__(self, index):
        return self.data[index]

    def __len__(self):
        return self.len

train_data = DataLoader(RandomDataset(32, 64), batch_size=2)
trainer = Trainer(
    limit_train_batches=2,
    max_epochs=1,
    enable_progress_bar=False,
    enable_model_summary=False,
    enable_checkpointing=False,
)
model = BoringModel()
trainer.fit(model, train_dataloaders=train_data)

ckpt_path = 'resume.ckpt'
trainer.save_checkpoint(ckpt_path)

new_model = BoringModel.load_from_checkpoint("resume.ckpt")
print(f"Resumed Epoch: {new_model.current_epoch}")
print(f"Resumed Step:  {new_model.global_step}")

ckpt = torch.load("resume.ckpt")
print(f"Checkpoint Epoch: {ckpt['epoch']}")
print(f"Checkpoint Step:  {ckpt['global_step']}")
```

### Outputs 

Output:
```
Epoch: 0 Global step: 0
Epoch: 0 Global step: 1
Resumed Epoch: 0
Resumed Step:  0
Checkpoint Epoch: 1
Checkpoint Step:  2
```

Expected output:
```
Epoch: 0 Global step: 0
Epoch: 0 Global step: 1
Resumed Epoch: 1
Resumed Step:  2
Checkpoint Epoch: 1
Checkpoint Step:  2
```

### Environment

```
* CUDA:
        - GPU:
        - available:         False
        - version:           11.3
* Packages:
        - numpy:             1.21.2
        - pyTorch_debug:     False
        - pyTorch_version:   1.11.0
        - pytorch-lightning: 1.7.0dev
        - tqdm:              4.64.0
* System:
        - OS:                Linux
        - architecture:
                - 64bit
                - ELF
        - processor:         x86_64
        - python:            3.8.13
```

cc @awaelchli @ananthsub @ninginthecloud @rohitgr7 @carmocca @justusschock

**Comment by rohitgr7** [comment]
`load_from_checkpoint` just initialized the model and load the weights. current epoch and global step are part of trainer, which are loaded when you call `trainer.fit(..., ckpt_path=...)`. Although you can inspect the current epoch and global_step by loading the checkpoint explicitly.

**Comment by carmocca** [comment]
Yes. Unfortunately, we cannot reload the checkpoint until the trainer is running so this is working as expected.

Closing, but feel free to ask any further questions

**Comment by jcowles** [comment]
this is a change from previous behavior, no?

**Comment by carmocca** [comment]
I believe this has been like this since the beginning or a very early version

**Comment by roedoejet** [comment]
Maybe I'm missing something here but using `trainer.fit(..., ckpt_path=...)` loses the ability to update the learning rate when fine-tuning (see https://github.com/Lightning-AI/lightning/issues/5339) and then if we instead use `model_obj = model.load_from_checkpoint(last_ckpt)` and `trainer.fit(model_obj, data)` to resume training, global_step and epoch are indeed 0 again. What is the proper way to resume training while changing hyperparameters such as learning rate?

**Comment by QuentinSchau** [comment]
@roedoejet, I finally found a trick for using `.fit()` method to resume training and changing some hyper-parameters such as learning rate. The fact is you need to change the checkpoint. You can do it by loading it and resave after modification. Here an example :

```
# I use Tensorboard logging

path = check_point_Path_Of_Your_Trained_Model

# Load checkpoint
ckpt = torch.load(path)

# You should adapt if you have several optimizers
# Change learning rate :
ckpt["optimizer_states"][0]['param_groups'][0]["lr"]=0.05
# Change initial learning rate :
ckpt["optimizer_states"][0]['param_groups'][0]["initial_lr"]=0.05
# Save the new checkpoint
torch.save(ckpt, path)

# Resume train with new checkpoint
trainer.fit(lightning_model,datamodule=data_module,ckpt_path=path)
```

# Here an example of the learning rate :

![learning rate](https://github.com/Lightning-AI/lightning/assets/111083887/14c03838-b967-4101-95d9-e382817fc8f3)

# Epochs : 

![epoch](https://github.com/Lightning-AI/lightning/assets/111083887/97941589-067c-4409-958b-8f300a8806e5)

**Comment by qixuema** [comment]
> ckpt_path=

I appreciate your response! It was really helpful in addressing my concerns.

**Comment by qixuema** [comment]
> `load_from_checkpoint` just initialized the model and load the weights. current epoch and global step are part of trainer, which are loaded when you call `trainer.fit(..., ckpt_path=...)`. Although you can inspect the current epoch and global_step by loading the checkpoint explicitly.

I appreciate your response! It was really helpful in addressing my concerns.

**Comment by marz869** [comment]
> Maybe I'm missing something here but using `trainer.fit(..., ckpt_path=...)` loses the ability to update the learning rate when fine-tuning (see #5339) and then if we instead use `model_obj = model.load_from_checkpoint(last_ckpt)` and `trainer.fit(model_obj, data)` to resume training, we can adjust the learning rate but global_step and epoch are indeed 0 again. What is the proper way to resume training while changing hyperparameters such as learning rate?

@roedoejet I have same problem, did you find any solution??

**Comment by roedoejet** [comment]
> > Maybe I'm missing something here but using `trainer.fit(..., ckpt_path=...)` loses the ability to update the learning rate when fine-tuning (see #5339) and then if we instead use `model_obj = model.load_from_checkpoint(last_ckpt)` and `trainer.fit(model_obj, data)` to resume training, we can adjust the learning rate but global_step and epoch are indeed 0 again. What is the proper way to resume training while changing hyperparameters such as learning rate?
> 
> @roedoejet I have same problem, did you find any solution??

no, unfortunately not
